### ISSUE TITLE
License isn't actually open source

### ISSUE THREAD

**Issue by rhofour** [DJ-TechTools/Midi_Fighter_Twister_Open_Source#3]
The license text reads:

>  Permission is hereby granted, free of charge, to any person owning or possessing 
>  a DJ Tech-Tools MIDI Fighter Twister Hardware Device to view and modify this source 
>  code for personal use. Person may not publish, distribute, sublicense, or sell 
>  the source code (modified or un-modified). Person may not use this source code 
>  or any diminutive works for commercial purposes.

From the Open Source Initiative's page:

> Open source licenses are licenses that comply with the Open Source Definition — in brief, they allow software to be freely used, modified, and shared.

The license included here seems to allow none of that. Without owning a Twister I'm not granted permission to modify the source code. Even if I do own one it seems I wouldn't be able to contribute in the normal way (a pull request) since that would involve distributing my modifications. Finally, not being a lawyer I'm not sure if the commercial restriction prevents me from using the modified source code in any kind of performance I might make money from.

Could you re-license this project with a proper open source license like the GPL?

**Comment by wunnation** [comment]
Hi Rhofour,
You are correct that this license is more of a 'Shared Source' model.
No the code will not be reclassified as GPL, as it is being supplied only for use by owners of this device.
The DJTT team is certainly willing to listen to update requests, and review your source code for possible submission into the code base.

"I'm not sure if the commercial restriction prevents me from using the modified source code in any kind of performance I might make money from."
I see where it is unclear regarding performances using modified firmware. I do not believe that this is intended to be restricted. I will work with the team to modify this text.

**Comment by rhofour** [comment]
> The DJTT team is certainly willing to listen to update requests, and review your source code for possible submission into the code base.

Am I correct in understanding that I would have to do this by privately emailing a patch (since sending a PR on Github involves putting my modified firmware somewhere public)?

The limitation on publishing and distributing modified firmware seems like it would discourage potential contributors.

As someone who works at a tech company all of the intellectual property I create is owned by my employer. Since they like to encourage open-source work they make it very easy for me to release open-source code I've written, but since this code isn't open-source I'd have a much harder time contributing. I believe people at other tech companies are in a similar boat.

Obviously how you license your code is entirely your decision and the Shared Source model is certainly better than completely closed source, but I hope you reconsider this.

**Comment by wunnation** [comment]
You bring up a good point there regarding contributions. The main purpose of providing this code is to allow users to make modifications and improvements, and the current license text does allow that.

However, I know that the DJTT team also wants to allow contributions from users. I'd imagine that the license needs to be modified to allow public posting in specific instances. I will bring this up.  Thanks for the input.

**Comment by Trinitou** [comment]
Hey @padi04 , is it allowed to create pull requests with actual code changes for this repo? The last posts here are a few years old and it seems like something makes people hesitate to contribute code. Can you say something concerning whether the license allows it or if not whether there are plans to make it more open for contributors?

**Comment by padi04** [comment]
@Trinitou absolutely! We are more than happy to have Twister be a platform you guys can tweak to your needs and come up with things that might not be in our radar.

The only thing we are against would be using Twister's firmware as a starting point for other hardware sold commercially.

We are planning on updating the license to clear up any confusion in it's current itineration and giving you guys the peace of mind to have fun tinkering with it.
